crates/bevy_math/src/curve/mod.rs

```diff
@@ -757,6 +882,171 @@ where
     }
 }
 
+/// The curve that results from reversing the current curve on the x-axis.
+///
+/// Curves of this type are produced by [`Curve::reverse`].
+pub struct ReverseCurve<T, C> {
+    curve: C,
+    _phantom: PhantomData<T>,
+}
+
+impl<T, C> Curve<T> for ReverseCurve<T, C>
+where
+    C: Curve<T>,
+{
+    #[inline]
+    fn domain(&self) -> Interval {
+        self.curve.domain()
+    }
+
+    #[inline]
+    fn sample_unchecked(&self, t: f32) -> T {
+        self.curve.sample_unchecked(self.domain().end() - t)
```

Comment:
I think this logic is not correct for non-zero `start`

For example this test would fail:

```rust

    #[test]
    fn reverse_with_negatives() {
        let curve = function_curve(Interval::new(-2.0, 1.0).unwrap(), |t| t * 3.0 + 1.0);
        let rev_curve = curve.reverse().unwrap();
        assert_eq!(rev_curve.sample_unchecked(-2.0), 1.0 * 3.0 + 1.0);
        assert_eq!(rev_curve.sample_unchecked(-0.5), -0.5 * 3.0 + 1.0);
        assert_eq!(rev_curve.sample_unchecked(1.0), -2.0 * 3.0 + 1.0);
    }
```
It would be need to replaced to something like:

```rust
self.curve.sample_unchecked(self.curve.domain().end() - (t - self.curve.domain().start()))
```